app/src/components/blocks/_lloydGeorge/lloydGeorgeViewRecordStage/LloydGeorgeViewRecordStage.tsx

```diff
@@ -61,20 +61,49 @@ function LloydGeorgeViewRecordStage({
     });
 
     useEffect(() => {
+        let historyIndex = window.history.state?.index || 0;
+
         if (fullScreen) {
-            window.history.pushState({ fullScreen: true }, '', window.location.href);
+            historyIndex += 1;
+            window.history.pushState(
+                { ...window.history.state, fullScreen: true, index: historyIndex },
+                '',
+                window.location.href,
+            );
         }
-        const handleBackButton = (event: Event) => {
-            if (fullScreen) {
+
+        const handleBrowserNavigationButton = (event: PopStateEvent) => {
+            const newIndex = event.state?.index || 0;
+
+            if (newIndex < historyIndex) {
                 event.preventDefault();
+                handleBrowserBackButton();
+            } else if (newIndex > historyIndex) {
+                event.preventDefault();
+                handleBrowserForwardButton();
+            }
+
+            historyIndex = newIndex;
+        };
+
+        window.addEventListener('popstate', handleBrowserNavigationButton);
+
+        const handleBrowserBackButton = () => {
+            if (fullScreen) {
                 setFullScreen(false);
+            } else {
+                window.history.back();
             }
         };
 
-        window.addEventListener('popstate', handleBackButton);
+        const handleBrowserForwardButton = () => {
+            if (!fullScreen) {
+                setFullScreen(true);
+            }
+        };
```

Comment:
Considering these functions are purely doing IF statements on fullscreen, could they not be combined into 1 function?

Reply:
They can, but they have been seperated like this to make it clear which one is to be done in back button click and which logic is to be done for forward button click

Reply:
Makes sense. I guess it's a matter of preference, whether having 2 functions justify the better readability or not. But I'm fine with that reasoning.